test/microbench/org/apache/cassandra/test/microbench/index/sai/memory/WriteBlockMemoryIndexBenchmark.java

```diff
@@ -0,0 +1,131 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.test.microbench.index.sai.memory;
+
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.cassandra.db.Clustering;
+import org.apache.cassandra.dht.Murmur3Partitioner;
+import org.apache.cassandra.index.sai.memory.MultiBlockIndex;
+import org.apache.cassandra.index.sai.memory.TrieMemoryIndex;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+@Fork(1)
+@Warmup(iterations = 5, time = 3)
+@Measurement(iterations = 10, time = 3)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@BenchmarkMode(Mode.AverageTime)
+@State(Scope.Thread)
+public class WriteBlockMemoryIndexBenchmark extends AbstractTrieMemoryIndexBenchmark
+{
+    //@Param({ "1000", "10000", "100000", "1000000" })
+    @Param({ "100000", "1000000" })
+    protected int numberOfTerms;
+
+    //@Param({ "1", "10", "100"})
+    @Param({ "1"})
+    protected int rowsPerPartition;
+
+    private MultiBlockIndex integerBlockIndex;
+
+    @Setup(Level.Iteration)
+    public void initialiseColumnData()
+    {
+        initialiseColumnData(numberOfTerms, rowsPerPartition);
+    }
+
+    @Setup(Level.Invocation)
+    public void initialiseIndexes()
+    {
+        //stringIndex = new TrieMemoryIndex(stringContext);
+        integerIndex = new TrieMemoryIndex(integerContext);
+
+        integerBlockIndex = new MultiBlockIndex(integerContext);
+    }
+
+//    @Benchmark
+//    public long writeStringIndex()
+//    {
+//        long size = 0;
+//        int rowCount = 0;
+//        int keyCount = 0;
+//        for (ByteBuffer term : stringTerms)
+//        {
+//            stringIndex.add(partitionKeys[keyCount], Clustering.EMPTY, term);
+//            if (++rowCount == rowsPerPartition)
+//            {
+//                rowCount = 0;
+//                keyCount++;
+//            }
+//            size++;
+//        }
+//        return size;
+//    }
```

Comment:
Keeping commented-out code is not a good idea. 
Commented code doesn't undergo any checking when compiling or linting.
It is also excluded when doing automatic refactoring so it can get easily broken.
You can always select a set of benchmarks to run by passing proper regex to JMH invocation.